src/Connections/GenericHook.ts

```diff
@@ -119,6 +146,29 @@ export class GenericHookConnection implements IConnection {
 
     }
 
+    public static getProvisionerDetails(botUserId: string) {
+        return {
+            service: "generic",
+            eventType: GenericHookConnection.CanonicalEventType,
+            type: "Webhook",
+            // TODO: Add ability to configure the bot per connnection type.
+            botUserId: botUserId,
+        }
+    }
+
+    public getProvisionerDetails() {
+        const url = `${this.config.urlPrefix}${this.config.urlPrefix.endsWith('/') ? '' : '/'}${this.hookId}`;
+        return {
+            ...GenericHookConnection.getProvisionerDetails(this.botUserId),
+            id: this.connectionId,
+            config: {
+                transformationFunction: this.transformationFunction,
+                hookId: this.hookId,
+                url,
+            },
```

Comment:
hook names would be great, if only for display purposes 😇 

I think this should be relatively straightforward, but a bit larger than a github suggestion can handle